test/functional/shared.js

```diff
@@ -57,8 +57,37 @@ function setupDatabase(configuration, dbsToClean) {
     );
 }
 
-function makeCleanupFn(client) {
-  return function(err) {
+function withTempDb(name, options, client, operation, errorHandler) {
```

Comment:
Similar to the change streams PR, I'd like to find a way not to wrap up `withClient` calls into something that doesn't seem to create a client. At very least I think we should have composed naming for these methods, like `withClientAndTempDb`, but maybe there's an even better way. Maybe we could end up with something like:
```
withClient(withTempDb((client, db, done) => {

}));
```

Reply:
I was actually working on this in a separate draft PR (#2362) so I pulled in my new `withDb` helper.